app/models/chargeback_vm.rb

```diff
@@ -83,7 +83,8 @@ def self.build_results_for_report_ChargebackVm(options)
   def self.where_clause(records, options, region)
     scope = records.where(:resource_type => "VmOrTemplate")
     if options[:tag] && (@report_user.nil? || !@report_user.self_service?)
-      scope.for_tag_names(options[:tag].split("/")[2..-1])
+      scope_with_current_tags = scope.where(:resource => Vm.find_tagged_with(:any => @options[:tag], :ns => '*'))
```

Comment:
It seems odd that we hardcode VmOrTemplate on line 84 and Vm on line 86.  This class should have knowledge what classes it's handling at a high level, such as a class method.  I'm not sure why it's VmOrTemplate above and Vm below.  Should it be `VmOrTemplate.find_tagged_with` ?

Reply:
yes it is -- but we are charging only VMs and not VmOrTemplate. So to have `VmOrTemplate ` here  is  basically useless.

But reason `VmOrTemplate` on line 84 is that we are not doing metric rollups for only type VM but for VmOrTemplate. 

but if you think that we have to keep it more consistent  I can change it to `VmOrTemplate.find_tagged_with ` but it is not needed.

Reply:
@lpichler we can address these hardcoded class names in a followup.  It feels like we need a method call to replace these class names:

```
class ChargebackVm < Chargeback
  ..
  def self.chargeable_resource
    Vm
  end
end
```

Then, all of these hardcoded places can call `chargeable_resource` or `chargeable_resource.to_s` or `chargeable_resource.tableize` depending if they need `Vm`, `"Vm"`, or "vms"

Reply:
Again, we don't have to do this now, it's just something I'm noticing while looking at these change...